### DISCUSSION TITLE
qimage producer issue with webp images

### DISCUSSION THREAD

**Discussion by ShrugerXu** [mltframework/mlt#1044] [Announcements]
Hello,

I'm having issue compositing webp images, the second webp image of a track ("2_webp_ll.webp", shows up at approximately the 4th second of the output video mentioned below) becomes a white chunk under Ubuntu **while normal under Windows**, here are my melt commands:
Ubuntu (melt 7.28.0 docker image):
```
/usr/bin/dumb-init -- /usr/bin/xvfb-run -a /usr/local/bin/melt -producer color:red out=500 -track -producer 1_webp_a.webp in=0 out=100 -producer 2_webp_ll.webp in=100 out=200 -transition composite -consumer avformat:test_linux_28.mp4 out=300
```
Windows (using melt.exe from Shotcut-240913):
```
melt.exe -producer color:red out=500 -track -producer 1_webp_a.webp in=0 out=100 -producer 2_webp_ll.webp in=100 out=200 -transition composite -consumer avformat:test_win.mp4 out=300
```

At first I thought it was because of different qimage version between the two os, so I tried to build melt using qt6 under Ubuntu 24, but it turned out the same, white chunk still appeared. 

I'm confused why there is such a difference, any advice would be very grateful.

_images I use:_
[images.zip](https://github.com/user-attachments/files/17530311/images.zip)

_output:_ 
```
Linux
https://github.com/user-attachments/assets/50085ec2-2b22-4753-b565-ba9c85c31358

WIndows
https://github.com/user-attachments/assets/99d322e2-b7ce-481d-9833-c090e85fc664

"white chunk" screenshot
https://github.com/user-attachments/assets/44915c63-df6f-46d9-9b23-76044a47059a
```

**Comment by bmatherly** [comment]
I performed a quick test on my Ubuntu 20.04.6 machine.

Here is what I did:
1) Download the shotcut portable zip from the [shotcut website](https://shotcut.org/download/)
2) Download your images linked above
3) Run the command (copied from your docker command):
`melt -producer color:red out=500 -track -producer 1_webp_a.webp in=0 out=100 -producer 2_webp_ll.webp in=100 out=200 -transition composite -consumer avformat:test_linux_28.mp4 out=300`

The resulting file correctly showed the penguin image at 4 seconds (no white chunk).

You are correct that we use Qt to decode the webp file. Maybe there is still a difference in the versions we are using.

You could try running the Shotcut portable melt command in your docker image as a test to rule out the environment.

**Reply by ShrugerXu** [reply]
Using shotcut portable works, I will do further version comparison.

Thank you very much!
